src/wallet/rpcwallet.cpp

```diff
@@ -3809,6 +3816,152 @@ UniValue getaddressinfo(const JSONRPCRequest& request)
     return ret;
 }
 
+/** Convert CAddressBookData to JSON record.
+ * The verbosity of the output is configurable based on the command.
+ */
+static UniValue AddressBookDataToJSON(const CAddressBookData& data, bool verbose)
+{
+    UniValue ret(UniValue::VOBJ);
+    if (verbose) {
+        ret.push_back(Pair("name", data.name));
+    }
+    ret.push_back(Pair("purpose", data.purpose));
+    if (verbose) {
+        UniValue ddata(UniValue::VOBJ);
+        for (const std::pair<std::string, std::string>& item : data.destdata) {
+            ddata.push_back(Pair(item.first, item.second));
+        }
+        ret.push_back(Pair("destdata", ddata));
+    }
+    return ret;
+}
+
+UniValue getlabel(const JSONRPCRequest& request)
+{
+    CWallet * const pwallet = GetWalletForJSONRPCRequest(request);
+    if (!EnsureWalletIsAvailable(pwallet, request.fHelp)) {
+        return NullUniValue;
+    }
+
+    if (request.fHelp || request.params.size() != 1)
+        throw std::runtime_error(
+            "getlabel \"bitcoinaddress\"\n"
+            "\nReturns the label associated with the given address.\n"
+            "\nArguments:\n"
+            "1. \"bitcoinaddress\"  (string, required) The bitcoin address for label lookup.\n"
+            "\nResult:\n"
+            "  { (json object with information about address)\n"
```

Comment:
> Can we make this RPC return an array?

It might be better to not add a `getlabel` RPC at all but instead just return this information in `getaddressinfo` (recently added in #10583).